### PR TITLE
Upgrade to bazel 5.2.0

### PR DESCRIPTION
**Pull request by tobijk**
Upgrade to 5.2.0. All patches apply cleanly, literally no changes required.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
+  sha256: 820a94dbb14071ed6d8c266cf0c080ecb265a5eea65307579489c4662c2d582a
   patches:
     - patches/0001-allow-args-to-be-passed-to-bazel_build.patch  # [unix]
     - patches/0002-do-not-use-enable-gold-unless-supported.patch  # [unix]
-    - patches/0003-Use-LSFindApplicationForInfo-for-locating-xcode.patch  # [osx]
     - patches/0004-link-against-iokit.patch  # [osx]
     - patches/0005-fix_netrc_protocol.patch  # [osx]
     - patches/0006-missing_inttype_h.patch
     - patches/0007-patch2.patch
-    - patches/0008-include-limits.patch
```
Comment (cbouss) [inline recipe/meta.yaml:18]: about/home: https://bazel.build/
about/doc_url: https://bazel.build/docs

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 
 build:
   number: 1
```
Comment (cbouss) [inline recipe/meta.yaml]: should be reset to 0

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,21 +1,19 @@
...
   version: {{ version }}
 
 source:
   url: https://github.com/bazelbuild/bazel/releases/download/{{ version }}/bazel-{{ version }}-dist.zip
-  sha256: 9981d0d53a356c4e87962847750a97c9e8054e460854748006c80f0d7e2b2d33
+  sha256: 820a94dbb14071ed6d8c266cf0c080ecb265a5eea65307579489c4662c2d582a
   patches:
     - patches/0001-allow-args-to-be-passed-to-bazel_build.patch  # [unix]
     - patches/0002-do-not-use-enable-gold-unless-supported.patch  # [unix]
-    - patches/0003-Use-LSFindApplicationForInfo-for-locating-xcode.patch  # [osx]
```
Comment (cbouss) [inline recipe/meta.yaml:13]: Asking the obvious, but I guess these were removed because of not being needed anymore.

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [inline recipe/meta.yaml]: Would we need to specify these deps?
https://github.com/bazelbuild/bazel/blob/master/MODULE.bazel#L12
I guess these are used? https://github.com/bazelbuild/bazel/tree/5.2.0/third_party

**Review by varlackc — COMMENTED** [review]
as soon as the changes mentioned take place, then we will be able to approve the pull request. 

need to reset the build number to zero

there is a redirect in the `home url`
https://bazel.build/

there is a redirection on the `doc_url`
https://bazel.build/install

File: recipe/meta.yaml (hunk 2 above)
Comment (tobijk) [reply recipe/meta.yaml]: My understanding is that Bazel is pretty much self-contained, once it is built, and it brings all its own dependencies in subprojects. Changing that would probably be a mammoth task.

Comment (tobijk) [reply recipe/meta.yaml:19]: I don't think this patch was ever strictly needed.

File: recipe/meta.yaml (hunk 3 above)
Comment (tobijk) [reply recipe/meta.yaml:13]: I don't think this patch was ever strictly needed and it doesn't work anymore.

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: That makes sense, thank you.

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**